tests/scene/test_path_follow_3d.h

```diff
@@ -32,188 +32,284 @@
 			"Progress should be clamped at 0");
 
 	path_follow_3d->set_progress(150);
 	CHECK_MESSAGE(
-			path_follow_3d->get_progress() == 100,
+			Math::is_equal_approx(path_follow_3d->get_progress(), 100),
 			"Progress should be clamped at max value of curve");
 
 	memdelete(path);
 }
+
+TEST_CASE("[SceneTree][PathFollow3D] Calculate forward vector") {
+	Ref<Curve3D> curve = memnew(Curve3D);
+	curve->add_point(Vector3(0, 0, 0));
+	curve->add_point(Vector3(100, 0, 0));
+	curve->add_point(Vector3(100, 100, 0));
+	curve->add_point(Vector3(100, 100, 100));
+	curve->add_point(Vector3(100, 0, 100));
+	Path3D *path = memnew(Path3D);
+	path->set_curve(curve);
+	PathFollow3D *path_follow_3d = memnew(PathFollow3D);
+	path->add_child(path_follow_3d);
+	SceneTree::get_singleton()->get_root()->add_child(path);
+
+	path_follow_3d->set_loop(false);
+	path_follow_3d->set_rotation_mode(PathFollow3D::RotationMode::ROTATION_ORIENTED);
+
+	path_follow_3d->set_progress(-50);
+	path_follow_3d->update_transform(true);
+	CHECK(is_equal_approx(Vector3(1, 0, 0), path_follow_3d->get_transform().get_basis().get_column(2)));
+
+	path_follow_3d->set_progress(0);
+	path_follow_3d->update_transform(true);
+	CHECK(is_equal_approx(Vector3(1, 0, 0), path_follow_3d->get_transform().get_basis().get_column(2)));
+
+	path_follow_3d->set_progress(50);
+	path_follow_3d->update_transform(true);
+	CHECK(is_equal_approx(Vector3(1, 0, 0), path_follow_3d->get_transform().get_basis().get_column(2)));
+
+	path_follow_3d->set_progress(100);
+	path_follow_3d->update_transform(true);
+	CHECK(is_equal_approx(Vector3(0, 1, 0), path_follow_3d->get_transform().get_basis().get_column(2)));
+
+	path_follow_3d->set_progress(150);
+	path_follow_3d->update_transform(true);
+	CHECK(is_equal_approx(Vector3(0, 1, 0), path_follow_3d->get_transform().get_basis().get_column(2)));
+
+	path_follow_3d->set_progress(200);
+	path_follow_3d->update_transform(true);
+	CHECK(is_equal_approx(Vector3(0, 0, 1), path_follow_3d->get_transform().get_basis().get_column(2)));
+
+	path_follow_3d->set_progress(250);
+	path_follow_3d->update_transform(true);
+	CHECK(is_equal_approx(Vector3(0, 0, 1), path_follow_3d->get_transform().get_basis().get_column(2)));
+
+	path_follow_3d->set_progress(300);
+	path_follow_3d->update_transform(true);
+	CHECK(is_equal_approx(Vector3(0, -1, 0), path_follow_3d->get_transform().get_basis().get_column(2)));
+
+	path_follow_3d->set_progress(350);
+	path_follow_3d->update_transform(true);
+	CHECK(is_equal_approx(Vector3(0, -1, 0), path_follow_3d->get_transform().get_basis().get_column(2)));
+
+	path_follow_3d->set_progress(400);
+	path_follow_3d->update_transform(true);
+	CHECK(is_equal_approx(Vector3(0, -1, 0), path_follow_3d->get_transform().get_basis().get_column(2)));
+
+	path_follow_3d->set_progress(450);
+	path_follow_3d->update_transform(true);
+	CHECK(is_equal_approx(Vector3(0, -1, 0), path_follow_3d->get_transform().get_basis().get_column(2)));
+
+	memdelete(path);
+}
```

Comment:
I rebuilt the unit test in the editor and tested it with 4.1.3, 4.2.2 and 4.3 - exact same behavior, which is actually expected for the oriented rotation mode.
However, I think we should consider renaming the items of the RotationMenu enum since they are a bit confusing in my opinion (at least `ROTATION_ORIENTED`), but that's for another PR.
I'd adjust the tests the following way to avoid target vectors becoming parallel to the up vector. The checks were verified using a test project:

![grafik](https://github.com/user-attachments/assets/c4afa94b-c61c-4faa-85ae-d4f8887e1b7b)


```suggestion
TEST_CASE("[SceneTree][PathFollow3D] Calculate forward vector") {
	const real_t dist_cube_100 = 100 * Math::sqrt(3.0);
	Ref<Curve3D> curve = memnew(Curve3D);
	curve->add_point(Vector3(0, 0, 0));
	curve->add_point(Vector3(100, 0, 0));
	curve->add_point(Vector3(200, 100, -100));
	curve->add_point(Vector3(200, 100, 200));
	curve->add_point(Vector3(100, 0, 100));
	curve->add_point(Vector3(0, 0, 100));
	Path3D *path = memnew(Path3D);
	path->set_curve(curve);
	PathFollow3D *path_follow_3d = memnew(PathFollow3D);
	path->add_child(path_follow_3d);
	SceneTree::get_singleton()->get_root()->add_child(path);

	path_follow_3d->set_loop(false);
	path_follow_3d->set_rotation_mode(PathFollow3D::RotationMode::ROTATION_ORIENTED);

	path_follow_3d->set_progress(-50);
	path_follow_3d->update_transform(true);
	CHECK(is_equal_approx(Vector3(-1, 0, 0), path_follow_3d->get_transform().get_basis().get_column(2)));

	path_follow_3d->set_progress(0);
	path_follow_3d->update_transform(true);
	CHECK(is_equal_approx(Vector3(-1, 0, 0), path_follow_3d->get_transform().get_basis().get_column(2)));

	path_follow_3d->set_progress(50);
	path_follow_3d->update_transform(true);
	CHECK(is_equal_approx(Vector3(-1, 0, 0), path_follow_3d->get_transform().get_basis().get_column(2)));

	path_follow_3d->set_progress(100);
	path_follow_3d->update_transform(true);
	CHECK(is_equal_approx(Vector3(-1, 0, 0), path_follow_3d->get_transform().get_basis().get_column(2)));

	path_follow_3d->set_progress(100 + dist_cube_100 / 2);
	path_follow_3d->update_transform(true);
	CHECK(is_equal_approx(Vector3(-0.577348, -0.577348, 0.577348), path_follow_3d->get_transform().get_basis().get_column(2)));

	path_follow_3d->set_progress(100 + dist_cube_100 - 0.01);
	path_follow_3d->update_transform(true);
	CHECK(is_equal_approx(Vector3(-0.577348, -0.577348, 0.577348), path_follow_3d->get_transform().get_basis().get_column(2)));

	path_follow_3d->set_progress(250 + dist_cube_100);
	path_follow_3d->update_transform(true);
	CHECK(is_equal_approx(Vector3(0, 0, -1), path_follow_3d->get_transform().get_basis().get_column(2)));

	path_follow_3d->set_progress(400 + dist_cube_100 - 0.01);
	path_follow_3d->update_transform(true);
	CHECK(is_equal_approx(Vector3(0, 0, -1), path_follow_3d->get_transform().get_basis().get_column(2)));

	path_follow_3d->set_progress(400 + 1.5 * dist_cube_100);
	path_follow_3d->update_transform(true);
	CHECK(is_equal_approx(Vector3(0.577348, 0.577348, 0.577348), path_follow_3d->get_transform().get_basis().get_column(2)));

	path_follow_3d->set_progress(400 + 2 * dist_cube_100 - 0.01);
	path_follow_3d->update_transform(true);
	CHECK(is_equal_approx(Vector3(0.577348, 0.577348, 0.577348), path_follow_3d->get_transform().get_basis().get_column(2)));

	path_follow_3d->set_progress(500 + 2 * dist_cube_100);
	path_follow_3d->update_transform(true);
	CHECK(is_equal_approx(Vector3(1, 0, 0), path_follow_3d->get_transform().get_basis().get_column(2)));

	memdelete(path);
}
```

Reply:
If you say these are the expected values I'll take your word for it.

Also, even though it's a bit off-topic for this PR, you mentioned renaming the enum members but that would break compatibility. We can however improve the documentation and the property hint that determines how they are displayed in the inspector.